src/researchhub_case/serializers/author_claim_case_serializer.py

```diff
@@ -6,65 +6,58 @@
 from researchhub_case.models import AuthorClaimCase
 from user.models import Author, User
 from user.serializers import AuthorSerializer, UserSerializer
-
+from django.contrib.contenttypes.models import ContentType
+from researchhub_case.tasks import (
+    trigger_email_validation_flow,
+)
 
 class AuthorClaimCaseSerializer(ModelSerializer):
     moderator = SerializerMethodField(method_name='get_moderator')
     requestor = SerializerMethodField()
-    target_author = SerializerMethodField(method_name='get_target_author')
+    paper = SerializerMethodField(method_name='get_paper')
 
     def create(self, validated_data):
         request_data = self.context.get('request').data
         moderator_id = request_data.get('moderator')
         requestor_id = request_data.get('requestor')
-        target_author_id = request_data.get('target_author')
+        target_paper_id = request_data.get('target_paper_id')
+        target_author_name = request_data.get('target_author_name')
         moderator = User.objects.filter(id=moderator_id).first()
         requestor = User.objects.filter(id=requestor_id).first()
-        author = request_data.get('author')
+
+        # An exception will be thrown if paper does not exist
+        Paper.objects.get(id=target_paper_id)
+
         self.__check_uniqueness_on_create(
             requestor_id,
-            target_author_id
+            target_paper_id,
+            target_author_name,
         )
 
-        if not target_author_id:
-            first_name = author.get('first_name')
-            last_name = author.get('last_name')
-            target_author = Author.objects.filter(
-                first_name=first_name,
-                last_name=last_name,
-                claimed=False
-            )
-
-            if target_author.exists():
-                target_author = target_author.first()
-            else:
-                target_author = Author.objects.create(
-                    first_name=first_name,
-                    last_name=last_name,
-                    claimed=False,
-                )
-
-                author_papers = Paper.objects.filter(
-                    raw_authors__contains=[
-                        {
-                            "first_name": first_name,
-                            "last_name": last_name
-                        }
-                    ]
-                )
-
-                for paper in author_papers:
-                    paper.authors.add(target_author)
-        else:
-            target_author = Author.objects.filter(id=target_author_id).first()
-
-        return AuthorClaimCase.objects.create(
+        case = AuthorClaimCase.objects.create(
             **validated_data,
+            target_paper_id=target_paper_id,
             moderator=moderator,
             requestor=requestor,
-            target_author=target_author
         )
 
+        trigger_email_validation_flow.apply_async((case.id,), priority=2)
```

Comment:
We need a short delay for db propagation. 
```suggestion
        trigger_email_validation_flow.apply_async(
            (case.id,),
            priority=2,
            countdown=5
        )
```

Reply:
Thanks. Make sense.